src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -232,8 +232,7 @@ std::shared_ptr<WorkerLeaseInterface>
 CoreWorkerDirectTaskSubmitter::GetOrConnectLeaseClient(
     const rpc::Address *raylet_address) {
   std::shared_ptr<WorkerLeaseInterface> lease_client;
-  if (raylet_address &&
-      NodeID::FromBinary(raylet_address->raylet_id()) != local_raylet_id_) {
+  if (NodeID::FromBinary(raylet_address->raylet_id()) != local_raylet_id_) {
```

Comment:
We always set `raylet_address` in `RequestNewWorkerIfNeeded` now, so the `nullptr` check is no longer needed.

Reply:
Can you add RAY_CHECK in this case? There's a possibility someone uses the method for other purposes and happen to write code that uses nullptr.

Reply:
I can add that `RAY_CHECK`, although I did manually check the repo and `GetOrConnectLeaseClient` isn't called anywhere other than in `RequestNewWorkerIfNeeded`, and it's a private method. But I suppose that doesn't guard very well against future development that might add or change a `GetOrConnectLeaseClient` call. Ideally this would change to a const reference, but I was going to wait until the locality-aware spillback PR to figure out that refactor.

I'll add the `RAY_CHECK` in the meantime!